### PR TITLE
Set OpenAPI maxLength on enum fields

### PR DESCRIPTION
**Pull request by cbandy**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?
   - [x] Have you added automated tests?

**Type of Changes:**

 
 - [x] Other

**What is the current behavior (link to any open issues here)?**

See: https://issue.k8s.io/119511

**What is the new behavior (if this is a feature change)?**

This helps old Kubernetes estimate reasonable costs for CEL validation rules.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -52,6 +52,10 @@ get-pgmonitor:
 	cp -r '$(PGMONITOR_DIR)/postgres_exporter/common/.' '${QUERIES_CONFIG_DIR}'
 	cp '$(PGMONITOR_DIR)/postgres_exporter/linux/queries_backrest.yml' '${QUERIES_CONFIG_DIR}'
 
+.PHONY: notes
+notes: ## List known issues and future considerations
+	@command -v rg > /dev/null && rg '(BUGS|FIXME|NOTE|TODO)[(][^)]+[)]' || grep -Ern '(BUGS|FIXME|NOTE|TODO)[(][^)]+[)]' *
```
Comment (tjmoore4) [inline Makefile:57]: :100: This is a nice way to keep track of these items.

File: pkg/apis/postgres-operator.crunchydata.com/v1beta1/crunchy_bridgecluster_types.go [hunk 2]
```diff
@@ -51,6 +51,11 @@ type CrunchyBridgeClusterSpec struct {
 
 	// The cloud provider where the cluster is located.
 	// Currently Bridge offers aws, azure, and gcp only
+	// ---
+	// Kubernetes assumes the evaluation cost of an enum value is very large.
+	// TODO(k8s-1.29): Drop MaxLength after Kubernetes 1.29; https://issue.k8s.io/119511
+	// +kubebuilder:validation:MaxLength=10
```
Comment (tjmoore4) [inline pkg/apis/postgres-operator.crunchydata.com/v1beta1/crunchy_bridgecluster_types.go:57]: :thinking: Just curious, why are some 10 and others 15? In both cases it seems to be several times larger than the number of enums listed.

**Comment by tjmoore4** [comment]
:memo: I wish I had caught this earlier, but since I've just merged this [PR](https://github.com/CrunchyData/postgres-operator/pull/4047), there is another set of enums that probably should be handled. If you can update this PR, that'd be great, otherwise I can put a PR together.

**Review by tjmoore4 — APPROVED** [review]

File: pkg/apis/postgres-operator.crunchydata.com/v1beta1/crunchy_bridgecluster_types.go (hunk 2 above)
Comment (cbandy) [reply pkg/apis/postgres-operator.crunchydata.com/v1beta1/crunchy_bridgecluster_types.go:57]: Oh, this is the max length of the _values_. The longest value on this field is `azure`, 5 characters, but I went higher so we wouldn't have to babysit the exact counts.

**Comment by cbandy** [comment]
> If you can update this PR, that'd be great, otherwise I can put a PR together.

👍🏻 I'll rebase and add it.

**Pull request merged**